### PR TITLE
Maintenance shroom rework and fix

### PR DESCRIPTION
**Pull request by Deon-DF**
## About The Pull Request

Before these changes, there are the following problems in the game:
- You can only sample maintenance fungus with wirecutters/scalpel. It does not use the tool qualities system.
- You get a single seed after you do that, which you cannot use until you grow it again.
- The seed will always grow in the same type of fungus with the same ingredient for the whole round, because the current botany system doesn't support inheriting reagents from parents, it checks the "name" in plant controller.

This PR does the following:
- Maintenance mushrooms get a name based on their reagent (randomized each round).
- That fixes the problem with all mushrooms having the same reagent after planting their seed.
- You can sample them with any cutting/wire cutting/laser cutting tool.
- You can sample either a single grown produce or a seed, via a radial menu selection.

radial

## Why It's Good For The Game

Now the fungi can actually net reagents they grow with, and they can be used without reusing them in the garden (and without rewriting the whole botany system to support reagent inheritance to seeds).

The custom fungi names are also kind of cute and seem fitting to the maintenance thematic.

## Changelog
:cl:
add: Added a random name generator for maintenance shrooms.
code: Reorganized maintenance mushroom reagent list to be alphabetical for easier reading.
fix: Cutting the maintenance plants now gives you a choice to either sample the plant (harvest 1) or sample the seed (harvest seed sample), with the radial menu selection.
fix: Sampling any maintenance plants can now be done with any tools that have "cutting", "wire cutting" or "laser cutting" qualities, instead of just wirecutters and scalpel.
/:cl:

### CODE REVIEW

**Comment by Deon-DF** [comment]
names

**Comment by Voldirs** [comment]
Could you please check if plants in hydroponics tray can be sampled (for seeds, not for product) too? It was possible to sample them before, it makes things much easier for xenobotany. And maybe revising possible chem list considering that we have more chems now?

**Comment by Deon-DF** [comment]
Voldirs, the change is that when you cut a growth with wirecutters, you get 1 product instead of 1 seed and that's the only change. Everything else is the same.

**Comment by Deon-DF** [comment]
> Also I'm not sure if shroom names should be capitalized
I understand the idea, I can rework it easily if I am asked of this by mergers, but tomorrow, it's 4 AM here.

**Comment by Deon-DF** [comment]
Example

Ahh screw it, 5 AM is not going to stop the progress.
I redid the names so they are no longer capitalized.

Also ^ here's a screenshot proving that my change didn't touch sampling the botany stuff from trays, it only affects plants growing in maintenance.

**Comment by Voldirs** [comment]
Sorry for taking your sleep, I didn't intend that!!!

**Comment by Deon-DF** [comment]
What about giving an option to either cut shroom or collect seed sample? That can be done with one menu.

**Comment by Deon-DF** [comment]
radial

**Comment by Deon-DF** [comment]
Ready for review now.

File: code/modules/hydroponics/seed_datums.dm [hunk 1]
```diff
@@ -551,57 +551,81 @@
...
-		"blattedin",
-		"honey",
-		"frostoil",
-		"capsaicin",
-		"banana",
-		"mutagen",
-		"chloralhydrate"
+		"sugar",
+		"synaptizine",
+		"thermite",
```
Comment (infraredbaron) [inline code/modules/hydroponics/seed_datums.dm:594]: Might melt walls into space, which is not really good.

**Pull request merged**